tests/components/folder/test_sensor.py

```diff
@@ -20,36 +17,33 @@ def create_file(path):
         test_file.write("test")
 
 
-class TestFolderSensor(unittest.TestCase):
-    """Test the filesize sensor."""
-
-    def setup_method(self, method):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        if not os.path.isdir(TEST_DIR):
-            os.mkdir(TEST_DIR)
-        self.hass.config.allowlist_external_dirs = {TEST_DIR}
-
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        if os.path.isfile(TEST_FILE):
-            os.remove(TEST_FILE)
-            os.rmdir(TEST_DIR)
-        self.hass.stop()
-
-    def test_invalid_path(self):
-        """Test that an invalid path is caught."""
-        config = {"sensor": {"platform": "folder", CONF_FOLDER_PATHS: "invalid_path"}}
-        assert setup_component(self.hass, "sensor", config)
-        assert len(self.hass.states.entity_ids()) == 0
-
-    def test_valid_path(self):
-        """Test for a valid path."""
-        create_file(TEST_FILE)
-        config = {"sensor": {"platform": "folder", CONF_FOLDER_PATHS: TEST_DIR}}
-        assert setup_component(self.hass, "sensor", config)
-        self.hass.block_till_done()
-        assert len(self.hass.states.entity_ids()) == 1
-        state = self.hass.states.get("sensor.test_folder")
-        assert state.state == "0.0"
-        assert state.attributes.get("number_of_files") == 1
+def remove_test_file():
+    """Remove test file."""
+    if os.path.isfile(TEST_FILE):
+        os.remove(TEST_FILE)
+        os.rmdir(TEST_DIR)
+
+
+async def test_invalid_path(hass):
+    """Test that an invalid path is caught."""
+    config = {"sensor": {"platform": "folder", CONF_FOLDER_PATHS: "invalid_path"}}
+    assert await async_setup_component(hass, "sensor", config)
+    assert len(hass.states.async_entity_ids()) == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)